### PR TITLE
feat-673: Manage eks addons

### PR DESCRIPTION
**Pull request by nightfury1204**
### What is the feature/fix?
https://github.com/convox/issues-private/issues/673

Rack update doesn't update the addons , so
Manage aws eks addons: vpc-cni, kube-proxy and coredns

### Does it has a breaking change?
**yes**
Once you upgrade to this version you can't downgrade, since addons are now managed by tf, so downgrading to lower version will delete those addons since they are not tf managed before and thus cluster breaks

### How to use/test it?
Install or upgrade your rack to this version

### Checklist
- [ ] New coverage tests
- [x] Unit tests passing
- [x] E2E tests passing
- [x] E2E downgrade/update test passing
- [ ] Documentation updated
- [x] No warnings or errors on Deepsource/Codecov

### CODE REVIEW

File: terraform/system/aws/main.tf [hunk 1]
```diff
@@ -60,6 +60,9 @@ module "cluster" {
   private             = var.private
   tags                = local.tag_map
   vpc_id              = var.vpc_id
+  vpc_cni_version = var.vpc_cni_version
```
Comment (yuriadams) [inline terraform/system/aws/main.tf:63]: just the `=`s are not aligned. The text editor should format this.

**Review by Twsouza — COMMENTED** [review]
I have some thoughts 🤔 

What terraform CLI version you used to test it? Did you get any `error: cycle` while updating?
(I just want to check if the error cycle was solved)

Is it possible to update the add-on for GKE/DO/other cloud providers?

Thanks!

one nitpick, please format the terraform files.

**Comment by nightfury1204** [comment]
> 



> I have some thoughts thinking
> 
> What terraform CLI version you used to test it? Did you get any `error: cycle` while updating? (I just want to check if the error cycle was solved)
> 
> Is it possible to update the add-on for GKE/DO/other cloud providers?
> 
> Thanks!
> 
> one nitpick, please format the terraform files.

@Twsouza
cli version locally in my pc `1.3.4`, but i also used the `convox rack update <version>` command. Also no i didn't get any cycle error

> Is it possible to update the add-on for GKE/DO/other cloud providers?

I didn't find anything explicit about that in terraform, probably need to look into that more and support it in another task

**Comment by Twsouza** [comment]
thanks, @nightfury1204 
could you create a ticket to look at the add-on for the other cloud providers?

File: terraform/cluster/aws/variables.tf [hunk 2]
```diff
@@ -10,6 +10,11 @@ variable "cidr" {
   default = "10.1.0.0/16"
 }
 
+variable "coredns_version" {
+  type = string
+  default = null
+}
```
Comment (Twsouza) [inline terraform/cluster/aws/variables.tf:16]: this file as well

**Comment by nightfury1204** [comment]
> thanks, @nightfury1204 could you create a ticket to look at the add-on for the other cloud providers?

done https://github.com/convox/issues-private/issues/714

**Review by Twsouza — APPROVED** [review]
LGTM

**Review by heronrs — APPROVED** [review]

**Pull request merged**